src/poetry/core/packages/path_dependency.py

```diff
@@ -0,0 +1,94 @@
+from __future__ import annotations
+
+import logging
+
+from abc import ABC
+from abc import abstractmethod
+from pathlib import Path
+from typing import Iterable
+
+from poetry.core.packages.dependency import Dependency
+from poetry.core.packages.utils.utils import path_to_url
+
+
+logger = logging.getLogger(__name__)
+
+
+class PathDependency(Dependency, ABC):
+    @abstractmethod
+    def __init__(
+        self,
+        name: str,
+        path: Path,
+        *,
+        source_type: str,
```

Comment:
how about `Literal['file'] | Literal['directory']` type or Enum?

Reply:
I'd like to use `Literal` but it's new in 3.8. Using Enum would probably entail a larger refactoring since we're using strings for all types of dependencies. I think that's out of scope for this PR. And introducing an Enum only for path dependencies is not worth it, at least in my opinion.